app/models/concerns/clone/session.rb

```diff
@@ -74,18 +74,18 @@ def matching_outcome_target_id(pattern, index)
     target_id = pattern['target'][index]['id']
     return check_if_session_exists(target_id) if pattern['target'][index]['type'].include?('Session') || target_id.empty?
 
-    matching_question_id(target_id)
+    matching_question(target_id)&.id || ''
   end
 
-  def matching_question_id(target_id)
+  def matching_question(target_id)
     target = check_if_question_exists(target_id)
     if target
       outcome.questions
              .joins(:question_group)
              .where(question_groups: { position: target.question_group.position })
-             .find_by!(position: target.position).id
+             .find_by!(position: target.position)
```

Comment:
I'm afraid that searching by position isn't correct anymore... For example, if we have HFHS initial screen we won't have in the copied session so the position won't match